src/components/AuthContainer.tsx

```diff
@@ -1,33 +1,60 @@
 import React, { useContext, useEffect } from 'react';
 import { SafeAreaView } from 'react-native';
 import { withAuthenticator, AmplifyTheme } from 'aws-amplify-react-native';
-import { Auth, I18n } from 'aws-amplify';
+import { Auth, I18n, API } from 'aws-amplify';
 import { Translations } from '@aws-amplify/ui-components';
+import { NavigationContainer } from '@react-navigation/native';
+import LoadingView from 'react-native-loading-view';
 import Onboarding from './Onboarding/Onboarding';
 import AuthenticatorTheme from '../styles/AuthenticatorTheme';
 import { AppContext } from '../contexts/AppContext';
 import NavFlow from './NavContainer';
+import { getUser } from '../graphql/queries';
+import { UserContext } from '../contexts/UserContext';
+import { QuestionProvider } from '../contexts/QuestionsContext';
 
 Auth.configure({ mandatorySignIn: true });
 
 function App(): JSX.Element | null {
   const { appState, setAppState } = useContext(AppContext);
+  const { userState, setUserState } = useContext(UserContext);
 
   useEffect(() => {
-    if (appState === 'Auth') {
-      setAppState('Onboarding'); // TODO: use datastore to check whether this user has at least one home
+    const getUserData = async () => {
+      const user = await Auth.currentAuthenticatedUser();
+      const result: any = await API.graphql({
+        query: getUser,
+        variables: { id: user.attributes.sub },
+      });
+      setUserState(result.data.getUser);
```

Comment:
nit: you can actually type `result` by doing:
```ts
const result = (await API.graphql({...})) as { data: GetUserQuery };
```
While typing `result` doesn't really help since it's only used once right below it, its still good convention to have explicit types whenever we can, especially since Amplify generated those types for us already.

Reply:
I have it as type `any` because typescript will throw an error when I set it in the context. This is because I am extracting the data and not including some additional objects in the context that AWS includes in a GraphQL response.